src/main/java/Deadline.java

```diff
@@ -0,0 +1,27 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Encapsulates information and state of a Deadline.
+ * For tasks that must be ccompletely by a certain time.
+ */
+public class Deadline extends Task {
+    /** Deadline of task. */
+    protected LocalDate by;
```

Comment:
I think you could change the name of this variable to make it more descriptive! Maybe something like "dueDate".